app/views/home/index.html.haml

```diff
@@ -53,14 +53,24 @@
           %th= heading_helper FinancialTransaction, :created_on
           %th= heading_helper FinancialTransaction, :user
           %th= heading_helper FinancialTransaction, :note
-          %th= heading_helper FinancialTransaction, :amount
+          %th= heading_helper FinancialTransaction, :deposit
+          %th= heading_helper FinancialTransaction, :memebership_fee
+          %th= heading_helper FinancialTransaction, :credit
         - for ft in current_user.ordergroup.financial_transactions.limit(5).order('created_on DESC')
           %tr
             %td= format_time(ft.created_on)
             %td= h(show_user(ft.user))
             %td= h(ft.note)
             - color = ft.amount < 0 ? 'red' : 'black'
-            %td{:style => "color:#{color}; width:5em", :class => "currency"}= number_to_currency(ft.amount)
+            %td.currency{:style => "color:#{color}; width:5em"}
+              - if ft.deposit?
+                = number_to_currency(ft.amount)
+            %td.currency{:style => "color:#{color}; width:5em"}
+              - if ft.membership_fee?
+                = number_to_currency(ft.amount)
+            %td.currency{:style => "color:#{color}; width:5em"}
+              - if ft.credit?
+                = number_to_currency(ft.amount)
```

Comment:
Hmm, I understand. And wonder if this can be simplified. The membership fee could be a debit of the account, either by a cronjob, or when placing an order.
The deposit is really different. I'd consider this a candidate for a plugin. One approach would be to debit the ordergroup's account with the deposit upon creation. Then when the member leaves, create a credit transaction of the deposit (which you can find by transaction description), and then pay back the remaining funds and create a financial transaction for that; then delete the ordergroup (or only remove its users). In this case you keep a full record, and don't need different "subaccounts". (One more improvement to introduce an ordergroup status in Foodsoft, so that members can become inactive, and the plugin could credit the deposit on their account on that transition.)
What would you think of this approach?